packages/react/src/table/TableScroller.tsx

```diff
@@ -0,0 +1,333 @@
+import {
+	type ReactNode,
+	useCallback,
+	useEffect,
+	useRef,
+	useState,
+} from 'react';
+import { Box } from '../box';
+import { boxPalette } from '../core';
+import { Flex } from '../flex';
+import { ScrollbarArrowLeftIcon, ScrollbarArrowRightIcon } from '../icon';
+import { Stack } from '../stack';
+
+export type TableScrollerProps = { children: ReactNode };
+
+export function TableScroller({ children }: TableScrollerProps) {
+	const trackRef = useRef<HTMLDivElement>(null);
+	const thumbRef = useRef<HTMLButtonElement>(null);
+	const scrollerRef = useRef<HTMLDivElement>(null);
+	const mousePos = useRef({ x: 0, y: 0 });
+
+	const [isDraggingThumb, setIsDraggingThumb] = useState(false);
+	const [thumbPosition, setThumbPosition] = useState(0);
+	const [thumbWidthRatio, setThumbWidthRatio] = useState(0);
+	const [buttonIntervalId, setButtonIntervalId] = useState<number | null>(null);
+
+	const repositionThumb = useCallback(() => {
+		if (!scrollerRef?.current) {
+			return;
+		}
+
+		setThumbPosition(scrollerRef.current.scrollLeft * thumbWidthRatio);
+	}, [scrollerRef, thumbWidthRatio]);
+
+	const calculateThumbWidth = useCallback(() => {
+		if (!scrollerRef?.current) {
+			return;
+		}
+
+		setThumbWidthRatio(
+			(scrollerRef?.current.offsetWidth - 56) /
+				(scrollerRef?.current.scrollWidth - 56)
+		);
+	}, [scrollerRef, setThumbWidthRatio]);
+
+	useEffect(() => {
+		calculateThumbWidth();
+	}, [calculateThumbWidth]);
+
+	useEffect(() => {
+		if (
+			typeof window === 'undefined' ||
+			!('ResizeObserver' in window) ||
+			!scrollerRef.current
+		) {
+			return;
+		}
+
+		const observer = new ResizeObserver(() => {
+			calculateThumbWidth();
+			repositionThumb();
+		});
+
+		observer.observe(scrollerRef.current);
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [calculateThumbWidth, repositionThumb, scrollerRef]);
+
+	const handleThumbPress = (event: React.MouseEvent | React.TouchEvent) => {
```

Comment:
Should this be wrapped in a `useCallback` as well? Otherwise it will be re-created every render but there's nothing in this function that I can see that updates with props/state changes. So, is this enough?

```ts
const handleThumbPress = useCallback((event: React.MouseEvent | React.TouchEvent) => {
    // ...
  },
  []
);
```

Reply:
Not sure. I would err on using as few `useCallback`s as possible (only where it complains), as the argument could be made for all of the functions.

Reply:
Yeah, fair point, I like that approach.